File: 백준/Silver/11047. 동전 0/동전 0.py
from sys import stdin

n, k = map(int, input().split())
coins = [int(stdin.readline()) for _ in range(n)][::-1]
result = 0

for coin in coins:
    result += k // coin
    k %= coin
    if k == 0:
        break
    
print(result)

File: 백준/Silver/11047. 동전 0/동전 0.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String nk[] = br.readLine().split(" ");
        int n  = Integer.parseInt(nk[0]), k = Integer.parseInt(nk[1]);
        Integer coins[] = new Integer[n];
        for(int i = 0 ; i < n ; i++) {
            coins[i] = Integer.parseInt(br.readLine());
        }
        Arrays.sort(coins, Collections.reverseOrder());

        int count = 0;
        for (int cost:coins) {
            count += k / cost;
            k %= cost;
        }

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        bw.write(String.valueOf(count));
        bw.flush();
    }
}